### ISSUE TITLE
Update `Publish-AWSPowerShellLambda` to set function runtime if it does not match the built csproj

### ISSUE THREAD

**Issue by mryanmurphy** [aws/aws-lambda-dotnet#1267]
### Describe the feature

`Publish-AWSPowerShellLambda` would be a better experience if it would automatically upgrade the function config to match the correct dotnet runtime prior to uploading function code. There is no parameter to specify this, so you need to trigger it via some other mechanism.

### Use Case

When publishing a function via `Publish-AWSPowerShellLambda`, if the current function config does not match what was built the deployment will fail. 

```
Error updating code for Lambda function: The runtime parameter of dotnetcore2.1 is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (dotnet6) while creating or updating functions.
```

### Proposed Solution

The csproj target framework is specified in blueprints: https://github.com/aws/aws-lambda-dotnet/blob/master/PowerShell/Module/Templates/Blueprints/projectfile.csproj.txt#L3

This should be used to call `dotnet lambda update-function-config` to set the runtime prior to updating function code.

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### AWS .NET SDK and/or Package version used

AWSLambdaPSCore 3.0.1.0

### Targeted .NET Platform

.NET 6

### Operating System and version

Windows 10

**Comment by ashishdhingra** [comment]
One of the caveat here is if the function runtime is actually supported by Lambda environment. For example, .NET Core 2.1 is no longer supported. Needs review with the team.

**Comment by ashishdhingra** [comment]
Fix should be similar for another related issue https://github.com/aws/aws-lambda-dotnet/issues/476.

**Comment by 96malhar** [comment]
This issue cannot be reproduced since it requires an existing Lambda function with a .NET runtime that is no longer supported by Lambda. The same problem was also reported to us for C# based Lambda functions - https://github.com/aws/aws-lambda-dotnet/issues/476.

As outlined by [normj's comment](https://github.com/aws/aws-lambda-dotnet/issues/476#issuecomment-1136551513), this issue was caused by the fact that Amazon.Lambda.Tools made 2 service calls -- the first call was to update the function code followed by a second call to update the function runtime.

We released a fix for this issue in Amazon.Lambda.Tools that flipped the order of these service calls - https://github.com/aws/aws-extensions-for-dotnet-cli/pull/265.

Since the `Publish-AWSPowerShellLambda` cmdlet uses Amazon.Lambda.Tools [under the covers,](https://github.com/aws/aws-lambda-dotnet/blob/f1c00fdc2b6c794e4dc222ca0143c0b74d107ffb/PowerShell/Module/Private/_DeploymentFunctions.ps1#L374) the above fix also applies to PowerShell based Lambda functions.
